doc/topics/best_practices.rst

```diff
@@ -21,6 +21,13 @@ General rules
 4. Store sensitive data in pillar.
 5. Don't use grains for matching in your pillar top file for any sensitive
    pillars.
+6. When accessing modules from within a template, use the mapping
+   key syntax instead of the attribute one to avoid edge cases. Example:
+
+   .. code-block:: jinja
+
+       {%- set do_this = salt['pillar.get']('foo:bar') %}
```

Comment:
Is this about the missing default value? Otherwise not sure how the `.get` comment above applies from here on. This example is not about accessing dict keys, but how to use Salt execution modules in Jinja templates in general. `salt.pillar.get` should be avoided, the same with `salt.cmd.run` etc. `salt["pillar.get"]` or `salt["cmd.run"]` don't have weird edge cases and should thus be preferred.

I'll answer the other disagrees here as well:

`salt["pillar.get"]` additionally traverses into subdicts, so the `pillar.get` version would look like this:

`{{ pillar.get("myapp", {}).get("version", "master") }}`

or

`{{ pillar | traverse("myapp:version", "master") }}`

Changing from `salt.pillar.get` to `salt["pillar.get"]` only changes the way the execution module is accessed, not other semantics.